cunumeric/module.py

```diff
@@ -523,6 +523,112 @@ def tile(a, reps):
     return result
 
 
+# Spliting arrays
+@copy_docstring(np.vsplit)
+def vsplit(a, indices):
+    return split(a, indices, axis=0)
+
+
+@copy_docstring(np.hsplit)
+def hsplit(a, indices):
+    return split(a, indices, axis=1)
+
+
+@copy_docstring(np.dsplit)
+def dsplit(a, indices):
+    return split(a, indices, axis=2)
+
+
+@copy_docstring(np.split)
+def split(a, indices, axis=0):
+    return array_split(a, indices, axis, equal=True)
+
+
+@copy_docstring(np.array_split)
+def array_split(a, indices, axis=0, equal=False):
+    array = ndarray.convert_to_cunumeric_ndarray(a)
+    dtype = type(indices)
+    split_pts = []
+
+    if dtype == int:
+        res = array.shape[axis] % indices
+        len_subarr = array.shape[axis] // indices
+        first_idx = len_subarr
+        end_idx = array.shape[axis]
+
+        if array.ndim >= axis:
+            if equal and array.shape[axis] % indices == 0:
+                raise ValueError(
+                    "array split does not result in an equal divison"
+                )
+        else:
+            raise ValueError(
+                "array({}) has less dimensions than axis({})".format(
+                    array.shape, axis
+                )
```

Comment:
I suggest we use f-strings exclusively in new code. We don't have to support python < 3.7.